### PR TITLE
docs: updates and new manual page for pcp-htop

### PR DESCRIPTION
**Pull request by natoscott**
Add some words about pcp-htop to the main man page, and add a
new man page describing the pcp-htop configuration files that
allow new meters and columns to be defined at runtime.

### CODE REVIEW

**Review by BenBE — COMMENTED** [review]
Wrapping single/few words to next line makes this unnecessarily hard to read.

File: pcp-htop.5.in [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+but metrics from other domains (\c
+.BR pmdabcc (1),
+.BR pmdabpf (1),
+etc) that have per-process values are equally applicable.
+.TP
+.B caption
+The short title that will be displayed at the head of the
+column \- usually a short, cryptic, all uppercase string.
+.TP
+.B description
```
Comment (smalinux) [inline pcp-htop.5.in:200]: I added also the "width" parameter. the width of the column users can control it.

File: pcp-htop.5.in [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+The following is an example configuration for a new column
+showing open file descriptors for each process:
+.LP
+.ft CW
+.nf
+.in +0.5i
+[fdcount]
+metric = proc.fd.count
+caption = FDCOUNT
+description = Open file descriptors
```
Comment (smalinux) [inline pcp-htop.5.in:162]:
```suggestion
description = Open file descriptors
width = 9
```

File: pcp-htop.5.in [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+or
+.I hotproc
+namespace (\c
+.BR pmdaproc (1)),
+but metrics from other domains (\c
+.BR pmdabcc (1),
+.BR pmdabpf (1),
+etc) that have per-process values are equally applicable.
+.TP
+.B caption
```
Comment (smalinux) [inline pcp-htop.5.in:196]: "color" parameter is missing?

File: pcp-htop.5.in (hunk 2 above)
Comment (BenBE) [reply pcp-htop.5.in:162]: What about `9` vs. `-9` for right and left alignment (similar to usage in format strings)?

Comment (smalinux) [reply pcp-htop.5.in:162]: @BenBE great 🙂 thank you. I thought to make left align for all Dynamic Columns. but this idea is better.

Comment (BenBE) [reply pcp-htop.5.in:162]: AFAIC this in the patchset you're passing that value through to `xSnprintf` anyway already …
But maybe there should be *some* sane limit in place …

**Comment by smalinux** [comment]
Related https://github.com/htop-dev/htop/pull/669 https://github.com/htop-dev/htop/pull/707 https://github.com/htop-dev/htop/pull/536

File: pcp-htop.5.in [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+A configuration file can contain multiple meter definitions.
+Each definition begins with a identifying name enclosed
+by square brackets \-
+.I redisclient
+in this example.
+The name is used internally within
+.B pcp-htop
+and must be unique, must begin with an alphabetic character,
+and may subsequently only contain alphanumeric characters or
+the underscore character.
```
[the remark this answers is not in the document]
Comment (natoscott) [reply pcp-htop.5.in:58]: As discussed elsewhere, dash is syntax for subtraction in the PCP derived metrics language we're using within pcp-htop.

File: pcp-htop.5.in [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+.B caption
+will be used for this.
+If neither is present, the internal (mandatory)
+.B name
+will be used.
+.TP
+.B type
+This setting allows a preferred default meter type to be
+specified.
+The associated value must be one of
```
[the remark this answers is not in the document]
Comment (natoscott) [reply pcp-htop.5.in:86]: Done.

File: pcp-htop.5.in [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+.TP
+.B name.label
+An optional, short label to display before the metric value.
+The ":" character will be appended to the
+.I label
+before the metric value part of the display.
+.TP
+.B name.suffix
+An optional, short suffix to display after the metric value.
+Commonly used to indicate values as a percentage using a "%"
```
[the remark this answers is not in the document]
Comment (natoscott) [reply pcp-htop.5.in:146]: Done.

File: pcp-htop.5.in (hunk 2 above)
Comment (natoscott) [reply pcp-htop.5.in:162]: I've added a 'width' section and discussion about alignment.

File: pcp-htop.5.in (hunk 3 above)
Comment (natoscott) [reply pcp-htop.5.in:196]: As discussed elsewhere, for column's we can use the regular htop palette and not allow customization here.  At least in this initial version - perhaps later we'll need more, but I suspect even then just setting one color wont provide the best functionality compared to what other columns can do, we may need some smarter logic.

File: pcp-htop.5.in [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+.LP
+A configuration file can contain multiple column definitions.
+Each definition begins with a identifying name enclosed
+by square brackets \-
+.I fdcount
+in this example, and the same rules apply as described above
+for meters.
+.LP
+Each column must specify a metric and optional parameters can
+also be set.
```
[the remark this answers is not in the document]
Comment (natoscott) [reply pcp-htop.5.in:175]: Done.

**Comment by BenBE** [comment]
@natoscott Does it make sense to merge this way before #707 lands?

File: pcp-htop.5.in [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+but metrics from other domains (\c
+.BR pmdabcc (1),
+.BR pmdabpf (1),
+etc) that have per-process values are equally applicable.
+.TP
+.B width
+Column width to use when displaying values for the metric.
+A negative value can be used to specify left alignment.
+An upper column limit of 28 characters is enforced.
+The default column width is 5 characters.
```
Comment (BenBE) [inline pcp-htop.5.in:207]: The code gives `-12` …

Comment (smalinux) [reply pcp-htop.5.in:207]: @BenBE since the default width +/-12 looks very large, I modified it to 5, thank you

**Comment by natoscott** [comment]
> @natoscott Does it make sense to merge this way before #707 lands?

It doesn't really matter once we're sure of the file formats, default values, and so on I guess.  We're getting close with #707 now, it's working pretty well for me and we have lots of useful additional columns.  So, either before or after should be fine from now on IMO - I don't have a strong preference.

**Comment by natoscott** [comment]
> Wrapping single/few words to next line makes this unnecessarily hard to read.

Happy to tweak things further but not sure what lines specifically this comment refers to... ?

File: pcp-htop.5.in [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+If not present, the
+.B caption
+will be used for this.
+If neither is present, the internal (mandatory)
+.B name
+will be used.
+.TP
+.B type
+This setting allows a preferred default meter type to be
+specified.
```
Comment (BenBE) [inline pcp-htop.5.in:89]: Really needed to word-wrap here?

File: pcp-htop.5.in [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+The associated value must be one of
+.IR bar ,
+.IR text ,
+.IR graph ,
+or
+.IR led .
+If no value is provided for a dynamic meter, the default
+value of
```
Comment (BenBE) [inline pcp-htop.5.in:97]: dito.

File: pcp-htop.5.in [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+Values sampled for each metric at runtime provide the
+animation visible in the
+.B pcp-htop
+display.
+The metric specification can be either a PCP metric name
+as listed by
+.BR pminfo (1)
+or a "derived" metric expression.
+The format for derived metric expressions is described
+on the
```
Comment (BenBE) [inline pcp-htop.5.in:128]: dito.

**Review by BenBE — COMMENTED** [review]
@natoscott Marked up some locations where there are only 1-3 words on a new line, where they technically didn't need to. Was worse in previous versions.

**Comment by natoscott** [comment]
> @natoscott Marked up some locations where there are only 1-3 words on a new line, where they technically didn't need to. Was worse in previous versions.

Ah, I see what you mean now.  I'll fix those cases you mention up and do a quick audit of the rest of the file - thanks.

File: pcp-htop.5.in [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+provide additional, optional meters and columns to extend the fixed
+set of display options provided by regular
+.BR htop .
+.LP
+These configuration files can be provided from both system-wide
+locations (first
+.I @sysconfdir@/pcp/htop
+then
+.IR @datarootdir@/pcp/htop )
+and below the users home directory (usually
```
Comment (BenBE) [inline pcp-htop.5.in:17]:
```suggestion
and below the user's home directory (usually
```

File: htop.1.in [hunk 13]
```diff
@@ -18,6 +22,19 @@ multiple processes and act on them all at once.
...
+.LP
+.B pcp-htop
+is a version of
+.B htop
+built using the Performance Co-Pilot (PCP) Metrics API (see \c
+.BR PCPIntro (1),
+.BR PMAPI (3)),
+allowing values not built into
+.B htop
+to be displayed from arbitrary metrics.
```
Comment (BenBE) [inline htop.1.in:34]:
```suggestion
allowing to extend
.B htop
to display values from arbitrary metrics.
```

File: pcp-htop.5.in [hunk 14]
```diff
@@ -0,0 +1,237 @@
...
+.IR ~/.config/htop ).
+Within these locations the
+.I meters
+and
+.I columns
+are scanned for dynamic Meter and Column specifications.
+.LP
+Meters are displayed in the top part of the
+.B pcp-htop
+window, and columns are display in the lower part.
```
Comment (BenBE) [inline pcp-htop.5.in:27]:
```suggestion
window, and columns are displayed in the lower part.
```

File: pcp-htop.5.in [hunk 15]
```diff
@@ -0,0 +1,237 @@
...
+Meters tend to display system-wide metrics, and Columns
+display metrics about individual processes.
+.LP
+The formats are similar but have slightly different requirements.
+Both formats follow the common ini-style, also allowing both empty
+lines and comment lines starting with the "#" character.
```
Comment (BenBE) [inline pcp-htop.5.in:33]:
```suggestion
Both formats follow the common ini-style. Blank lines are ignored.
Lines starting with the "#" character are treated as comments.
```

**Review by BenBE — APPROVED** [review]
Just some small changes left …

**Pull request merged**